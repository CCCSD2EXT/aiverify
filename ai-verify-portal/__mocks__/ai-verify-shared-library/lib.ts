export function getComponents() {
  return {};
}

export function parseRJSFSchema() {
  return {
    sensitive_feature: {
      items: {},
    },
  };
}
