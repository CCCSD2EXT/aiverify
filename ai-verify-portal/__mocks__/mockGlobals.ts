function silentConsoleLogs() {
  jest.spyOn(console, 'log').mockImplementation(jest.fn());
  jest.spyOn(console, 'debug').mockImplementation(jest.fn());
  jest.spyOn(console, 'warn').mockImplementation(jest.fn());
  jest.spyOn(console, 'error').mockImplementation(jest.fn());
}

export { silentConsoleLogs };
