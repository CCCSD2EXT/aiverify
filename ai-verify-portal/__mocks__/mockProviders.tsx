import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { NotificationsProvider } from 'src/modules/notifications/providers/notificationsContext';
import { PropsWithChildren } from 'react';

type MockProvidersProps = {
  apolloMocks?:
    | MockedResponse<Record<string, any>, Record<string, any>>[]
    | undefined;
};

function MockProviders(props: PropsWithChildren<MockProvidersProps>) {
  const { apolloMocks, children } = props;
  return (
    <MockedProvider mocks={apolloMocks} addTypename={false}>
      <NotificationsProvider>{children}</NotificationsProvider>
    </MockedProvider>
  );
}

export { MockProviders };
