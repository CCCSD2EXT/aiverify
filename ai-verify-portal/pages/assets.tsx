import AssetsModule from 'src/modules/assets';

/**
 * Home page
 */
export default function AssetsPage() {
  return <AssetsModule />;
}
