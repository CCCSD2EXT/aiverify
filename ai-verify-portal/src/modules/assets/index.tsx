import { useRouter } from 'next/router';

import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Paper from '@mui/material/Paper';
import Grid from '@mui/material/Unstable_Grid2';
import Container from '@mui/material/Container';

import NavigateBeforeIcon from '@mui/icons-material/NavigateBefore';
import FolderIcon from '@mui/icons-material/Folder';
import AddIcon from '@mui/icons-material/Add';
import InsertDriveFileIcon from '@mui/icons-material/InsertDriveFile';
import HomeMenuButtonComponent from '../home/homeMenuButton';
import { MinimalHeader } from '../home/header';

import styles from './styles/new-asset.module.css';

/**
 * Assets page component
 */
export default function AssetsModule() {
  const router = useRouter();

  return (
    <div>
      <MinimalHeader />
      <div className="layoutContentArea">
        <Container maxWidth={false} className="scrollContainer">
          <Container maxWidth="xl" className="mainContainer">
            <Paper
              sx={{
                p: 2,
                m: 2,
                display: 'flex',
                flexDirection: 'column',
                height: '80vh',
              }}>
              <Box sx={{ height: '5%', display: 'flex', alignItems: 'center' }}>
                <Button
                  startIcon={<NavigateBeforeIcon />}
                  color="secondary"
                  sx={{ mr: 1 }}
                  onClick={() => router.push('/home')}
                  data-testid="assets-back-button"
                />
                <Typography
                  variant="h6"
                  sx={{ pl: 2, flexGrow: 1, fontWeight: 'bold' }}>
                  Manage Assets
                </Typography>
              </Box>
              <Box sx={{ p: 10, height: '95%' }}>
                <Grid container spacing={2} sx={{ mb: 1 }}>
                  <Grid xs={4}>
                    <div
                      className={styles.newButton}
                      onClick={() => router.push('/assets/newDataset')}
                      data-testid="add-new-dataset-button">
                      <div
                        style={{
                          display: 'flex',
                          flexDirection: 'row',
                          gap: '20px',
                          height: '30px',
                          alignItems: 'center',
                        }}>
                        <InsertDriveFileIcon></InsertDriveFileIcon>
                        <div style={{ fontSize: '18px', fontWeight: '500' }}>
                          New Dataset
                        </div>
                      </div>
                      <AddIcon></AddIcon>
                    </div>
                  </Grid>
                  <Grid xs={4}>
                    <div
                      className={styles.newButton}
                      onClick={() => router.push('/assets/newModel')}
                      data-testid="add-new-model-button">
                      <div
                        style={{
                          display: 'flex',
                          flexDirection: 'row',
                          gap: '20px',
                          height: '30px',
                          alignItems: 'center',
                        }}>
                        <InsertDriveFileIcon></InsertDriveFileIcon>
                        <div style={{ fontSize: '18px', fontWeight: '500' }}>
                          New AI Model
                        </div>
                      </div>
                      <AddIcon></AddIcon>
                    </div>
                  </Grid>
                </Grid>
                <Typography
                  variant="h6"
                  sx={{ pb: 2, fontWeight: 'bold', marginTop: '50px' }}>
                  Select Folder to open
                </Typography>
                <Grid container spacing={2} sx={{ mb: 1 }}>
                  <Grid xs={4}>
                    <HomeMenuButtonComponent
                      testid="open-dataset-list-button"
                      title="Datasets"
                      description="Datasets for testing, ground-truth & background"
                      onClick={() => router.push('/assets/datasets')}
                      MyIcon={FolderIcon}
                    />
                  </Grid>
                  <Grid xs={4}>
                    <HomeMenuButtonComponent
                      testid="open-model-list-button"
                      title="AI Models"
                      description="AI Models & Pipelines to be tested" //AI Models, Pipelines & API Configs
                      onClick={() => router.push('/assets/models')}
                      MyIcon={FolderIcon}
                    />
                  </Grid>
                </Grid>
              </Box>
            </Paper>
          </Container>
        </Container>
      </div>
    </div>
  );
}
