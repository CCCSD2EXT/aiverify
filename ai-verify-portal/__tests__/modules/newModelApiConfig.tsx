import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MockProviders } from '__mocks__/mockProviders';
import { silentConsoleLogs } from '__mocks__/mockGlobals';
import { NewModelApiConfigModule } from 'src/modules/assets/modelAPIComponent';
import * as apolloClient from '@apollo/client';

describe('New Model API Config', () => {
  beforeAll(() => {
    silentConsoleLogs();
  });

  describe('Initial Render', () => {
    it('should render with the correct defaults', async () => {
      const { container } = render(
        <MockProviders>
          <NewModelApiConfigModule />
        </MockProviders>
      );

      expect(
        await screen.findByText(/^Create API Configuration$/i)
      ).toBeInTheDocument();
      const selectedTabBtn = container.querySelector(
        '.tabBtn__selected'
      ) as Element;
      expect(selectedTabBtn.textContent).toBe('Request Body');
      expect(container).toMatchSnapshot('New Model API Config Default');
    });
  });

  describe('GraphQL Model API Config Payloads', () => {
    const addNewModelAPIConfigSpy = jest.fn();
    beforeAll(() => {
      jest.mock('@apollo/client', () => ({
        useMutation: jest.fn(() => [addNewModelAPIConfigSpy]),
      }));
    });

    it('should create the correct payload', async () => {
      const { container } = render(
        <MockProviders>
          <NewModelApiConfigModule />
        </MockProviders>
      );

      await screen.findByText(/^Create API Configuration$/i);
      const editNameBtn = await screen.findByTestId('editConfigIconBtn');
      const addPropertyBtn = await screen.findByTestId('addRequestPropertyBtn');
      const saveBtn = await screen.findByText(/^SAVE$/i);
      userEvent.click(editNameBtn);
      await screen.findByText(/^OK$/i);
      await userEvent.type(
        container.querySelector('input[name="name"]') as HTMLInputElement,
        'My test API'
      );
      await userEvent.type(
        container.querySelector('input[name="name"]') as HTMLInputElement,
        'My test API description'
      );
      await userEvent.type(
        container.querySelector('input[name="modelType"]') as HTMLInputElement,
        'Regression'
      );
      await userEvent.type(
        container.querySelector(
          'input[name="modelAPI.url"]'
        ) as HTMLInputElement,
        'https://localhost:5000/predict/tc001'
      );
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'age'
      );
      await userEvent.click(addPropertyBtn);
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'gender'
      );
      await userEvent.click(addPropertyBtn);
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'race'
      );
      await userEvent.click(addPropertyBtn);
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'income'
      );
      await userEvent.click(addPropertyBtn);
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'employment'
      );
      await userEvent.click(addPropertyBtn);
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'employment_length'
      );
      await userEvent.click(addPropertyBtn);
      await userEvent.type(
        container.querySelector('input[name="paramName"]') as HTMLInputElement,
        'num_donation'
      );
      await userEvent.click(addPropertyBtn);
      await waitFor(async () => {
        expect(
          Array.from(container.querySelectorAll('.propertyInputRow')).length
        ).toEqual(8);
      });

      await userEvent.click(saveBtn);
      await waitFor(async () => {
        expect(addNewModelAPIConfigSpy).toHaveBeenCalled();
      });
    }, 10000);
  });
});
