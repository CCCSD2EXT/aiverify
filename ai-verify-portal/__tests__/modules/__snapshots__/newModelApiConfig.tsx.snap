// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Model API Config Initial Render should render with the correct defaults: New Model API Config Default 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        style="cursor: pointer;"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
      <img
        alt="AI Verify"
        class="logo"
        data-nimg="1"
        decoding="async"
        height="240"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="240"
      />
      <div
        style="position: absolute; right: 20px;"
      >
        <div
          class="root"
        >
          <div
            class="icon"
          >
            <button
              class="iconBtn iconBtn__noOutline aiv_iconbtn"
              type="button"
            >
              <div
                class="btnContent"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="NotificationsNoneIcon"
                  focusable="false"
                  style="font-size: 25px; color: rgb(120, 120, 120);"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
                  />
                </svg>
                <div
                  style="display: flex; align-items: center;"
                />
              </div>
            </button>
          </div>
          <div
            class="flyOutMsg"
          />
        </div>
      </div>
    </div>
    <div
      class="layoutContentArea"
    >
      <div
        class="scrollContainer"
      >
        <div
          class="mainContainer"
        >
          <div
            class="container__limits"
          >
            <div
              class="layout"
            >
              <form
                action="#"
              >
                <div
                  style="margin-bottom: 25px;"
                >
                  <h3
                    class="screenHeading"
                  >
                    Create API Configuration
                  </h3>
                  <p
                    class="headingDescription"
                  >
                    Create a new API configuration needed to connect to the AI model server
                  </p>
                </div>
                <div
                  class="pageLevelError"
                />
                <div
                  class="apiConfigForm"
                >
                  <div
                    class="leftSection"
                  >
                    <div>
                      <div
                        style="display: flex; flex-direction: column;"
                      >
                        <div
                          style="display: flex; align-items: flex-start;"
                        >
                          <div
                            class="configName"
                          >
                            Configuration Name
                          </div>
                          <button
                            class="iconBtn iconBtn__noOutline aiv_iconbtn"
                            data-testid="editConfigIconBtn"
                            id="editConfigIconBtn"
                            style="font-size: 12px;"
                            type="button"
                          >
                            <div
                              class="btnContent"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="EditIcon"
                                focusable="false"
                                style="font-size: 20px; color: rgb(120, 120, 120);"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                                />
                              </svg>
                              <div
                                style="display: flex; align-items: center;"
                              />
                            </div>
                          </button>
                        </div>
                        <div>
                          <div
                            class="description"
                          >
                            Description
                          </div>
                        </div>
                      </div>
                      <div
                        style="margin-top: 25px;"
                      >
                        <div
                          class="selectInput"
                          id="modelTypeSelect"
                          style="width: auto;"
                        >
                          <label
                            for="modelType"
                          >
                            <div
                              class="label"
                            >
                              <div>
                                Model Type
                              </div>
                            </div>
                            <div
                              class=" css-fyq6mk-container"
                            >
                              <span
                                class="css-1f43avz-a11yText-A11yText"
                                id="react-select-2-live-region"
                              />
                              <span
                                aria-atomic="false"
                                aria-live="polite"
                                aria-relevant="additions text"
                                class="css-1f43avz-a11yText-A11yText"
                              />
                              <div
                                class="aiv__control css-l8cqhx-control"
                              >
                                <div
                                  class="aiv__value-container aiv__value-container--has-value css-1tkr00k-ValueContainer"
                                >
                                  <div
                                    class="aiv__single-value css-1dimb5e-singleValue"
                                  >
                                    Classification
                                  </div>
                                  <div
                                    class="aiv__input-container css-12a5c88-Input"
                                    data-value=""
                                  >
                                    <input
                                      aria-autocomplete="list"
                                      aria-expanded="false"
                                      aria-haspopup="true"
                                      autocapitalize="none"
                                      autocomplete="off"
                                      autocorrect="off"
                                      class="aiv__input"
                                      id="react-select-2-input"
                                      role="combobox"
                                      spellcheck="false"
                                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                      tabindex="0"
                                      type="text"
                                      value=""
                                    />
                                  </div>
                                </div>
                                <div
                                  class="aiv__indicators css-1hb7zxy-IndicatorsContainer"
                                >
                                  <span
                                    class="aiv__indicator-separator css-1j9nsr8-indicatorSeparator"
                                  />
                                  <div
                                    aria-hidden="true"
                                    class="aiv__indicator aiv__dropdown-indicator css-ogl6pu-indicatorContainer"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class="css-tj5bde-Svg"
                                      focusable="false"
                                      height="20"
                                      viewBox="0 0 20 20"
                                      width="20"
                                    >
                                      <path
                                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                      />
                                    </svg>
                                  </div>
                                </div>
                              </div>
                              <input
                                name="modelType"
                                type="hidden"
                                value="Classification"
                              />
                            </div>
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="vDivider"
                  />
                  <div
                    class="rightSection"
                  >
                    <div
                      style="display: flex;"
                    >
                      <div
                        style="margin-right: 10px;"
                      >
                        <div
                          class="selectInput"
                          style="width: 140px;"
                        >
                          <label
                            for="modelAPI.method"
                          >
                            <div
                              class="label"
                            >
                              <div>
                                Request Method
                              </div>
                            </div>
                            <div
                              class=" css-fyq6mk-container"
                            >
                              <span
                                class="css-1f43avz-a11yText-A11yText"
                                id="react-select-3-live-region"
                              />
                              <span
                                aria-atomic="false"
                                aria-live="polite"
                                aria-relevant="additions text"
                                class="css-1f43avz-a11yText-A11yText"
                              />
                              <div
                                class="aiv__control css-l8cqhx-control"
                              >
                                <div
                                  class="aiv__value-container aiv__value-container--has-value css-1tkr00k-ValueContainer"
                                >
                                  <div
                                    class="aiv__single-value css-1dimb5e-singleValue"
                                  >
                                    POST
                                  </div>
                                  <div
                                    class="aiv__input-container css-12a5c88-Input"
                                    data-value=""
                                  >
                                    <input
                                      aria-autocomplete="list"
                                      aria-expanded="false"
                                      aria-haspopup="true"
                                      autocapitalize="none"
                                      autocomplete="off"
                                      autocorrect="off"
                                      class="aiv__input"
                                      id="react-select-3-input"
                                      role="combobox"
                                      spellcheck="false"
                                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                      tabindex="0"
                                      type="text"
                                      value=""
                                    />
                                  </div>
                                </div>
                                <div
                                  class="aiv__indicators css-1hb7zxy-IndicatorsContainer"
                                >
                                  <span
                                    class="aiv__indicator-separator css-1j9nsr8-indicatorSeparator"
                                  />
                                  <div
                                    aria-hidden="true"
                                    class="aiv__indicator aiv__dropdown-indicator css-ogl6pu-indicatorContainer"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class="css-tj5bde-Svg"
                                      focusable="false"
                                      height="20"
                                      viewBox="0 0 20 20"
                                      width="20"
                                    >
                                      <path
                                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                      />
                                    </svg>
                                  </div>
                                </div>
                              </div>
                              <input
                                name="modelAPI.method"
                                type="hidden"
                                value="POST"
                              />
                            </div>
                          </label>
                        </div>
                      </div>
                      <div
                        style="flex-grow: 1;"
                      >
                        <div
                          class="textInput"
                        >
                          <label>
                            <div
                              class="label"
                            >
                              <div>
                                Model URL
                              </div>
                            </div>
                            <input
                              name="modelAPI.url"
                              type="text"
                              value=""
                            />
                          </label>
                        </div>
                      </div>
                    </div>
                    <div
                      class="tabs"
                    >
                      <div
                        class="tabsHeading"
                      >
                        HTTP Request and Connection Settings
                      </div>
                      <div
                        class="tabsBtnGroup"
                      >
                        <div
                          class="tabBtn tabBtn__selected"
                        >
                          Request Body
                        </div>
                        <div
                          class="tabBtn"
                        >
                          Additional Request Headers
                        </div>
                        <div
                          class="tabBtn"
                        >
                          Response Properties
                        </div>
                        <div
                          class="tabBtn"
                        >
                          Authentication Settings
                        </div>
                        <div
                          class="tabBtn"
                        >
                          Connection Settings
                        </div>
                      </div>
                      <div
                        class="tabsDivider"
                      />
                      <div
                        class="tabContent"
                      >
                        <div
                          style="display: flex; flex-direction: column;"
                        >
                          <div
                            class="selectInput"
                            style="width: 240px;"
                          >
                            <label
                              for="modelAPI.requestBody.mediaType"
                            >
                              <div
                                class="label"
                              >
                                <div>
                                  Media Type
                                </div>
                              </div>
                              <div
                                class=" css-fyq6mk-container"
                              >
                                <span
                                  class="css-1f43avz-a11yText-A11yText"
                                  id="react-select-4-live-region"
                                />
                                <span
                                  aria-atomic="false"
                                  aria-live="polite"
                                  aria-relevant="additions text"
                                  class="css-1f43avz-a11yText-A11yText"
                                />
                                <div
                                  class="aiv__control css-l8cqhx-control"
                                >
                                  <div
                                    class="aiv__value-container aiv__value-container--has-value css-1tkr00k-ValueContainer"
                                  >
                                    <div
                                      class="aiv__single-value css-1dimb5e-singleValue"
                                    >
                                      application/x-www-form-urlencoded
                                    </div>
                                    <div
                                      class="aiv__input-container css-12a5c88-Input"
                                      data-value=""
                                    >
                                      <input
                                        aria-autocomplete="list"
                                        aria-expanded="false"
                                        aria-haspopup="true"
                                        autocapitalize="none"
                                        autocomplete="off"
                                        autocorrect="off"
                                        class="aiv__input"
                                        id="react-select-4-input"
                                        role="combobox"
                                        spellcheck="false"
                                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                        tabindex="0"
                                        type="text"
                                        value=""
                                      />
                                    </div>
                                  </div>
                                  <div
                                    class="aiv__indicators css-1hb7zxy-IndicatorsContainer"
                                  >
                                    <span
                                      class="aiv__indicator-separator css-1j9nsr8-indicatorSeparator"
                                    />
                                    <div
                                      aria-hidden="true"
                                      class="aiv__indicator aiv__dropdown-indicator css-ogl6pu-indicatorContainer"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="css-tj5bde-Svg"
                                        focusable="false"
                                        height="20"
                                        viewBox="0 0 20 20"
                                        width="20"
                                      >
                                        <path
                                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                        />
                                      </svg>
                                    </div>
                                  </div>
                                </div>
                                <input
                                  name="modelAPI.requestBody.mediaType"
                                  type="hidden"
                                  value="application/x-www-form-urlencoded"
                                />
                              </div>
                            </label>
                          </div>
                          <div
                            style="display: flex; flex-direction: column; align-items: flex-start; margin-bottom: 15px;"
                          >
                            <div
                              class="checkbox"
                              style="margin-bottom: 15px;"
                            >
                              <label
                                class="control"
                                style="height: 20px; font-size: 20px;"
                              >
                                <input
                                  class="input"
                                  name="modelAPI.requestBody.isArray"
                                  type="checkbox"
                                  value=""
                                />
                              </label>
                              <div
                                class="labelText"
                              >
                                <div>
                                  Format as array
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            style="display: flex; margin-bottom: 4px;"
                          >
                            <div
                              class="headingName"
                            >
                              Property Name
                            </div>
                            <div
                              class="headingVal"
                            >
                              Data Type
                            </div>
                          </div>
                          <div
                            style="display: flex; flex-direction: column;"
                          >
                            <div
                              class="keyValRow propertyInputRow"
                            >
                              <div
                                class="keyValCol"
                              >
                                <div
                                  class="textInput"
                                  style="margin-bottom: 0px;"
                                >
                                  <label>
                                    <input
                                      maxlength="100"
                                      name="paramName"
                                      type="text"
                                      value=""
                                    />
                                  </label>
                                </div>
                              </div>
                              <div
                                class="keyValCol"
                              >
                                <div
                                  class="selectInput"
                                  style="width: auto; margin-bottom: 0px;"
                                >
                                  <label
                                    for="propDataType"
                                  >
                                    <div
                                      class=" css-fyq6mk-container"
                                    >
                                      <span
                                        class="css-1f43avz-a11yText-A11yText"
                                        id="react-select-5-live-region"
                                      />
                                      <span
                                        aria-atomic="false"
                                        aria-live="polite"
                                        aria-relevant="additions text"
                                        class="css-1f43avz-a11yText-A11yText"
                                      />
                                      <div
                                        class="aiv__control css-l8cqhx-control"
                                      >
                                        <div
                                          class="aiv__value-container aiv__value-container--has-value css-1tkr00k-ValueContainer"
                                        >
                                          <div
                                            class="aiv__single-value css-1dimb5e-singleValue"
                                          >
                                            integer
                                          </div>
                                          <div
                                            class="aiv__input-container css-12a5c88-Input"
                                            data-value=""
                                          >
                                            <input
                                              aria-autocomplete="list"
                                              aria-expanded="false"
                                              aria-haspopup="true"
                                              autocapitalize="none"
                                              autocomplete="off"
                                              autocorrect="off"
                                              class="aiv__input"
                                              id="react-select-5-input"
                                              role="combobox"
                                              spellcheck="false"
                                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                              tabindex="0"
                                              type="text"
                                              value=""
                                            />
                                          </div>
                                        </div>
                                        <div
                                          class="aiv__indicators css-1hb7zxy-IndicatorsContainer"
                                        >
                                          <span
                                            class="aiv__indicator-separator css-1j9nsr8-indicatorSeparator"
                                          />
                                          <div
                                            aria-hidden="true"
                                            class="aiv__indicator aiv__dropdown-indicator css-ogl6pu-indicatorContainer"
                                          >
                                            <svg
                                              aria-hidden="true"
                                              class="css-tj5bde-Svg"
                                              focusable="false"
                                              height="20"
                                              viewBox="0 0 20 20"
                                              width="20"
                                            >
                                              <path
                                                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                              />
                                            </svg>
                                          </div>
                                        </div>
                                      </div>
                                      <input
                                        name="propDataType"
                                        type="hidden"
                                        value="integer"
                                      />
                                    </div>
                                  </label>
                                </div>
                              </div>
                              <div
                                class="iconContainer"
                              >
                                <button
                                  class="iconBtn iconBtn__disabled aiv_iconbtn"
                                  data-testid="addRequestPropertyBtn"
                                  id="addRequestPropertyBtn"
                                  type="button"
                                >
                                  <div
                                    class="btnContent"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                      data-testid="AddIcon"
                                      focusable="false"
                                      style="font-size: 20px; color: rgb(120, 120, 120);"
                                      viewBox="0 0 24 24"
                                    >
                                      <path
                                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                                      />
                                    </svg>
                                    <div
                                      style="display: flex; align-items: center;"
                                    >
                                      <div
                                        style="color: rgb(103, 103, 103); font-size: 15px; margin: 0px 6px;"
                                      >
                                        Add
                                      </div>
                                    </div>
                                  </div>
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="buttons"
                >
                  <button
                    class="aivBase-button aivBase-button--secondary aivBase-button--medum"
                    style="width: 180px;"
                    type="button"
                  >
                    Back to AI Models
                  </button>
                  <div>
                    <button
                      class="aivBase-button aivBase-button--primary aivBase-button--medium"
                      style="width: 100px; margin-right: 0px;"
                      type="submit"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
